test/e2e/util.go

```diff
@@ -509,3 +531,32 @@ func mustGetEnv(name string) string {
 	}
 	return val
 }
+
+// updateInfrastructureStatus updates the Infrastructure status by applying
+// the given update function to the current Infrastructure object.
+// If there is a conflict error on update then the complete operation
+// is retried until timeout is reached.
+func updateInfrastructureConfigStatusWithRetryOnConflict(t *testing.T, timeout time.Duration, kubeClient client.Client, updateFunc func(*configv1.Infrastructure) *configv1.Infrastructure) error {
+	t.Helper()
+
+	t.Log("Updating 'cluster' infrastructure config status")
+	return wait.PollUntilContextTimeout(context.Background(), 5*time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+		infra := &configv1.Infrastructure{}
+		if err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infra); err != nil {
+			t.Logf("error getting 'cluster' infrastructure config: %v, retrying...", err)
+			return false, nil
+		}
+
+		// Apply the update function to the Infrastructure object.
+		updatedInfra := updateFunc(infra.DeepCopy())
```

Comment:
Interesting that we create `infra` as a pointer then we do a deep copy of the full object, change it and copy back to this function. Why not simply pass `infra` and change it inside `updateFunc`? Therefore avoiding the deep copy and a return copy (the latter will be avoid by RVO but still).

Reply:
Thanks for the suggestion. Now, I am passing `infra` directly into `updateFunc` without doing a deep copy.